src/components/molecules/Modal/Modal.jsx

```diff
@@ -0,0 +1,55 @@
+import './Modal.scss';
+import { useState, useEffect, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import { Icon } from '../../atoms/IconComponent/Icon';
+
+export const Modal = ({ children }) => {
```

Comment:
This component should be reusable, so, you are keeping the modal state in component itself, what if for example I would like to put a modal on feedScreen, and open it on onClick of for example "Edit profile" button, I can't do that with current implementation

Reply:
To do that we can just pass the state and setState as a props, and use them from props, in that way we would have state in parent component and we will be able to do anything we want with modal visibility